projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+// Comments ----------
+// I write a lot of comments, it has helped me the entire course to learn, understand and remember
+// They can all be removed at the end to clean it up
+
+func handleOKResponse(response *http.Response) {
+	// try to read the response body
+	responseBody, err := io.ReadAll(response.Body)
+
+	// use the defer keyword early to close the body regardless
+	defer response.Body.Close()
+
+	if err != nil {
+		// fmt.Fprintln(os.Stderr, "Parsing Response Body Failed. Error:", err)
+		// deliberately without explicit error message
+		fmt.Fprintln(os.Stderr, "Parsing Response Body Failed.")
+		return
+	}
+
+	// deliberately using os.Stdout (not fmt.Println) even though unnecessary (for the sake of the exercise)
+	fmt.Fprintln(os.Stdout, string(responseBody))
+}
+
+// note: returns an integer: 0 for fail || > 0 for success
+func parseRetryAfterHeader(retryAfterHeaderString string) int {
+	// first condition "a while"
+	if retryAfterHeaderString == "a while" {
+		return 0
+	}
+
+	// second condition an integer in string format
+	retryAfterInteger, err := strconv.Atoi(retryAfterHeaderString)
+	// if we successfully parse an integer return it
+	if err == nil {
+		return retryAfterInteger
+	}
+
+	// third condition a timestamp in string format
+	retryAfterTimeStamp, err := time.Parse(time.RFC1123, retryAfterHeaderString)
+	// if we cannot parse the timestamp then exit
```

Comment:
this comment says it will exit, but doesn't :) (I don't think it should exit here btw)